src/tests/unit/cpp_wrapper/test_cpp_wrapper.cpp

```diff
@@ -112,6 +113,439 @@ test_cpp_wrapper_insert_on_all_implementations(
+	test_cpp_wrapper_insert_update_edge_cases(tc, dict);
+	delete dict;
+}
+
+void
+test_cpp_wrapper_equality(
+		planck_unit_test_t	*tc,
+		Dictionary<int, int> *dict
+) {
+	int eq_key = 10;
+	int record_value = 0;
+
+	for(int i=0; i<eq_key; i++) {
+		dict->insert(eq_key, i);
+		PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+
+		dict->insert(i, i*2);
+		PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+	}
+
+	Cursor<int, int> *eq_cursor = dict->equality(eq_key);
+	PLANCK_UNIT_ASSERT_TRUE(tc, eq_cursor->hasNext());
+	eq_cursor->next();
+
+	while(eq_cursor->hasNext()) {
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, eq_key, eq_cursor->getKey());
+		PLANCK_UNIT_ASSERT_TRUE(tc, eq_cursor->getValue() == record_value);
+		PLANCK_UNIT_ASSERT_TRUE(tc, eq_cursor->hasNext());
+		eq_cursor->next();
+		record_value++;
+	}
+
+	PLANCK_UNIT_ASSERT_TRUE(tc, !eq_cursor->hasNext());
+
+	/* Check that same number of records are found as were inserted with desired key. */
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, eq_key, record_value);
+	delete eq_cursor;
+}
+
+void
+test_cpp_wrapper_equality_on_all_implementations(
+		planck_unit_test_t *tc
+) {
+	Dictionary<int, int> *dict;
+
+	dict = new BppTree<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int));
+	test_cpp_wrapper_equality(tc, dict);
+	delete dict;
+
+	dict = new SkipList<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int), 7);
+	test_cpp_wrapper_equality(tc, dict);
+	delete dict;
+
+//	dict = new FlatFile<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int));
+//	test_cpp_wrapper_equality(tc, dict);
+//	delete dict;
+
+//	dict = new OpenAddressHash<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int), 50);
+//	test_cpp_wrapper_equality(tc, dict);
+//	delete dict;
+
+//	dict = new OpenAddressFileHash<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int), 50);
+//	test_cpp_wrapper_equality(tc, dict);
+//	delete dict;
+}
+
+void
+test_cpp_wrapper_range(
+		planck_unit_test_t	*tc,
+		Dictionary<int, int> *dict
+) {
+	int min_key = 1;
+	int max_key = 5;
+
+	for(int i=0; i<10; i++) {
+		dict->insert(i, i);
+		PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+	}
+
+	Cursor<int, int> *range_cursor = dict->range(min_key, max_key);
+	PLANCK_UNIT_ASSERT_TRUE(tc, range_cursor->hasNext());
+	range_cursor->next();
+
+	while(range_cursor->hasNext()) {
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, min_key, range_cursor->getKey());
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, range_cursor->getValue(), min_key);
+		PLANCK_UNIT_ASSERT_TRUE(tc, range_cursor->hasNext());
+		range_cursor->next();
+		min_key++;
+	}
+
+	PLANCK_UNIT_ASSERT_TRUE(tc, !range_cursor->hasNext());
+	PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 5, --min_key);
```

Comment:
Should subtract prior to the assert call and **document** why this is necessary. Otherwise it looks spooky.